espnet/nets/chainer_backend/e2e_asr_transformer.py

```diff
@@ -260,31 +232,112 @@ def forward(self, xs, ilens, ys, calculate_attentions=False):
+                y_hats, ys_pad, self.char_list, self.space, self.blank)
+
+            wer = 0.0 if not self.report_wer else float(sum(word_eds)) / sum(word_ref_lens)
+            cer = 0.0 if not self.report_cer else float(sum(char_eds)) / sum(char_ref_lens)
+
+        # Print Output
+        if chainer.config.train and (self.verbose > 0 and self.char_list is not None):
+            for i, y_hat in enumerate(y_hats):
+                y_true = ys_pad[i]
+                if i == MAX_DECODER_OUTPUT:
+                    break
+                eos_true = np.where(y_true == -1)[0]
+                eos_true = eos_true[0] if len(eos_true) > 0 else len(y_true)
+                seq_hat = [self.char_list[int(idx)] for idx in y_hat[:eos_true]]
+                seq_true = [self.char_list[int(idx)] for idx in y_true[y_true != -1]]
+                seq_hat = "".join(seq_hat).replace(self.space, ' ')
+                seq_true = "".join(seq_true).replace(self.space, ' ')
+                logging.info("groundtruth[%d]: " % i + seq_true)
+                logging.info("prediction [%d]: " % i + seq_hat)
+
         alpha = self.mtlalpha
-        loss_ctc = None
-        if alpha == 0:
+        if alpha == 0.0:
             self.loss = loss_att
-        elif alpha == 1:
-            self.loss = None  # WIP
+            loss_att_data = loss_att.data
+            loss_ctc_data = None
+        elif alpha == 1.0:
+            self.loss = loss_ctc
+            loss_att_data = None
+            loss_ctc_data = loss_ctc.data
         else:
             self.loss = alpha * loss_ctc + (1 - alpha) * loss_att
-        if not math.isnan(self.loss.data):
-            reporter.report({'loss_ctc': loss_ctc}, self)
-            reporter.report({'loss_att': loss_att}, self)
+            loss_att_data = loss_att.data
+            loss_ctc_data = loss_ctc.data
+        loss_data = self.loss.data
+
+        if not math.isnan(loss_data):
+            reporter.report({'loss_ctc': loss_ctc_data}, self)
+            reporter.report({'loss_att': loss_att_data}, self)
             reporter.report({'acc': acc}, self)
 
-            logging.info('mtl loss:' + str(self.loss.data))
-            reporter.report({'loss': self.loss}, self)
+            reporter.report({'cer_ctc': cer_ctc}, self)
+            reporter.report({'cer': cer}, self)
+            reporter.report({'wer': wer}, self)
+
+            logging.info('mtl loss:' + str(loss_data))
+            reporter.report({'loss': loss_data}, self)
         else:
-            logging.warning('loss (=%f) is not correct', self.loss.data)
+            logging.warning('loss (=%f) is not correct', loss_data)
+
         if self.flag_return:
             loss_ctc = None
             return self.loss, loss_ctc, loss_att, acc
         else:
             return self.loss
 
-    def recognize(self, x_block, recog_args, char_list=None, rnnlm=None):
+    def recognize_beam2(self, x_block, recog_args, char_list=None, rnnlm=None, use_jit=False):
```

Comment:
Is this method is necessary?
I think this PR does not use `recognize_beam2` in other codes.

Reply:
I left this on purpose. Just in case the recognize_beam didnot work, but I will be removing before merge.

Reply:
I see, thank you.